File: SDKs/airbyte.local/config/1.0.0/python/src/sdk/models/shared/airbytestream.py
import dataclasses
from typing import Any,Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import syncmode_enum as shared_syncmode_enum


@dataclass_json
@dataclasses.dataclass
class AirbyteStream:
    r"""AirbyteStream
    the immutable schema defined by the source
    """
    
    name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    default_cursor_field: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('defaultCursorField') }})
    json_schema: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('jsonSchema') }})
    namespace: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('namespace') }})
    source_defined_cursor: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('sourceDefinedCursor') }})
    source_defined_primary_key: Optional[list[list[str]]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('sourceDefinedPrimaryKey') }})
    supported_sync_modes: Optional[list[shared_syncmode_enum.SyncModeEnum]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('supportedSyncModes') }})
    

File: SDKs/airbyte.local/config/1.0.0/java/lib/src/main/java/openapisdk/models/shared/AirbyteStream.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * AirbyteStream
 * the immutable schema defined by the source
**/
public class AirbyteStream {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("defaultCursorField")
    public String[] defaultCursorField;
    public AirbyteStream withDefaultCursorField(String[] defaultCursorField) {
        this.defaultCursorField = defaultCursorField;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("jsonSchema")
    public java.util.Map<String, Object> jsonSchema;
    public AirbyteStream withJsonSchema(java.util.Map<String, Object> jsonSchema) {
        this.jsonSchema = jsonSchema;
        return this;
    }
    @JsonProperty("name")
    public String name;
    public AirbyteStream withName(String name) {
        this.name = name;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("namespace")
    public String namespace;
    public AirbyteStream withNamespace(String namespace) {
        this.namespace = namespace;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("sourceDefinedCursor")
    public Boolean sourceDefinedCursor;
    public AirbyteStream withSourceDefinedCursor(Boolean sourceDefinedCursor) {
        this.sourceDefinedCursor = sourceDefinedCursor;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("sourceDefinedPrimaryKey")
    public String[][] sourceDefinedPrimaryKey;
    public AirbyteStream withSourceDefinedPrimaryKey(String[][] sourceDefinedPrimaryKey) {
        this.sourceDefinedPrimaryKey = sourceDefinedPrimaryKey;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("supportedSyncModes")
    public SyncModeEnum[] supportedSyncModes;
    public AirbyteStream withSupportedSyncModes(SyncModeEnum[] supportedSyncModes) {
        this.supportedSyncModes = supportedSyncModes;
        return this;
    }
}